core/pkg/artifacts/downloader.go

```diff
@@ -100,50 +223,68 @@ func (ad *ArtifactDownloader) downloadFiles(artifactID string, manifest Manifest
 	manifestEntries := manifest.Contents
 	numInProgress, numDone := 0, 0
 	nameToScheduledTime := map[string]time.Time{}
-	taskResultsChan := make(chan TaskResult)
+	taskResultsChan := make(chan TaskResult, MAX_BACKLOG)
```

Comment:
I made this buffered for the purpose of the tests, but I don't think it should affect how it runs in production